File: core/evidence/evidence_algebra_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def combine_evidence(evidence: List[str], weights: Dict[str, float] | None = None) -> Dict[str, Any]:
    """Deterministic evidence union with additive weight."""
    w = weights or {}
    items = sorted(set(str(e) for e in evidence if e))
    total = round(sum(w.get(e, 1.0) for e in items), 3)
    return {
        "items": items,
        "count": len(items),
        "weight_sum": total,
        "sufficient": len(items) >= 2,
        "deterministic_inputs": [f"count={len(items)}", f"weight_sum={total}"],
    }

File: src/evidence/evidenceAlgebraEngine.ts
/**
 * Converted from Python: core/evidence/evidence_algebra_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function combineEvidence(evidence: any, weights: any = null): any {
  var w: any = py.or2(weights, () => ({}));
  var items: any = py.sorted(py.toSet(py.iter(evidence).filter((e: any) => py.truthy(e)).map((e: any) => py.toStr(e))));
  var total: any = py.round(py.sum(py.iter(items).map((e: any) => py.get(w, e, py.F(1.0)))), 3);
  return {"items": items, "count": py.len(items), "weight_sum": total, "sufficient": (py.len(items) >= 2), "deterministic_inputs": [`count=${py.toStr(py.len(items))}`, `weight_sum=${py.toStr(total)}`]};
}
